java/insert-loadgen/src/main/java/org/kududb/examples/loadgen/InsertLoadgen.java

```diff
@@ -89,52 +94,20 @@ public static void runLoad(String masterHost, String tableName) {
 
       KuduSession session = client.newSession();
       session.setFlushMode(SessionConfiguration.FlushMode.AUTO_FLUSH_BACKGROUND);
-      while (true) {
+      for (int insertCount = 0; ; insertCount++) {
         Insert insert = table.newInsert();
         PartialRow row = insert.getRow();
         for (int i = 0; i < schema.getColumnCount(); i++) {
           generators.get(i).generateColumnData(row);
         }
         session.apply(insert);
-      }
 
-    } catch (Exception e) {
-      e.printStackTrace();
-    } finally {
-      try {
-        client.shutdown();
-      } catch (Exception e) {
-        e.printStackTrace();
-      }
-    }
-  }
-
-  public static void main(String[] args) {
-    if (args.length != 3) {
-      System.err.println("Usage: InsertLoadgen kudu_master_host kudu_table num_threads");
-      System.exit(1);
-    }
-
-    final String masterHost = args[0];
-    final String tableName = args[1];
-    int numThreads = Integer.parseInt(args[2]);
-
-    final CountDownLatch latch = new CountDownLatch(numThreads);
-
-    List<Thread> threads = new ArrayList<>(numThreads);
-    for (int i = 0; i < numThreads; i++) {
-      threads.add(new Thread(new Runnable() {
-        public void run() {
-          runLoad(masterHost, tableName);
-          latch.countDown();
+        if (insertCount % 1000 == 0) {
```

Comment:
Could check session.countPendingErrors first since getPendingErrors creates an array whether there are errors or not.